cmd/testsuite/cmd/root.go

```diff
@@ -0,0 +1,131 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/signal"
+	"path/filepath"
+	"syscall"
+	"time"
+
+	"github.com/spf13/cobra"
+
+	"github.com/G-Research/armada/internal/testsuite"
+	"github.com/G-Research/armada/pkg/client"
+)
+
+// RootCmd is the root Cobra command that gets called from the main func.
+// All other sub-commands should be registered here.
+func RootCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "testsuite",
+		Short: "testsuite is a suite of automated tests for Armada deployments.",
+		Long: `testsuite is a suite of automated tests for Armada deployments.
+
+Persistent config can be saved in a config file so it doesn't have to be specified every command.
+
+Example structure:
+armadaUrl: localhost:50051
+basicAuth:
+username: user1
+password: password123
+
+The location of this file can be passed in using the --config argument.
+If not provided, $HOME/.armadactl.yaml is used.`}
+
+	client.AddArmadaApiConnectionCommandlineArgs(cmd)
+
+	cmd.AddCommand(
+		versionCmd(testsuite.New()),
+		testCmd(testsuite.New()),
+	)
+
+	return cmd
+}
+
+// Print version info and exit.
+func versionCmd(app *testsuite.App) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "version",
+		Short: "Print version.",
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			return initParams(cmd, app)
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return app.Version()
+		},
+	}
+	return cmd
+}
+
+// Submit batches of jobs and wait for those jobs to finish.
+// Prints job completion statistics on exit.
+func testCmd(app *testsuite.App) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "test",
+		Short: "Test an Armada deployment by submitting jobs and watching for expected events.",
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			return initParams(cmd, app)
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+
+			testFilesPattern, err := cmd.Flags().GetString("tests")
+			if err != nil {
+				return err
+			}
+
+			testFiles, err := filepath.Glob(testFilesPattern)
+			if err != nil {
+				return err
+			}
+
+			// Crate a context that is cancelled on SIGINT/SIGTERM.
+			// Ensures test jobs are cancelled on ctrl-C.
+			ctx, cancel := context.WithCancel(context.Background())
```

Comment:
maybe not for now but we should definitely make a utility function for this

Reply:
I prefer not having a utility function. I think this is more clear.